ratik/src/main/java/org/opencadc/ratik/ArtifactValidator.java

```diff
@@ -322,9 +532,44 @@ protected void validateLocalAndRemote(Artifact local, Artifact remote) {
                 // evidence: local artifact has older Entity.lastModified indicating an update to
                 //           optional metadata at remote
                 // action: put Artifact
-                log.info(String.format("resolve Artifact.metaChecksum mismatch: put remote %s %s "
-                                           + "reason: remote lastModified newer than local",
-                                       remote.getID(), remote.getURI()));
+                try {
+                    log.debug("starting transaction");
+                    this.transactionManager.startTransaction();
+                    log.debug("start txn: OK");
+
+                    this.artifactDAO.lock(local);
+                    Artifact current = this.artifactDAO.get(local.getID());
+                    if (!current.getMetaChecksum().equals(remote.getMetaChecksum())) {
+                        if (current.getLastModified().before(remote.getLastModified())) {
+                            log.info(String.format("resolve Artifact.metaChecksum mismatch: put remote %s %s "
+                                                       + "reason: remote lastModified newer than local",
+                                                   remote.getID(), remote.getURI()));
+                            this.artifactDAO.put(remote);
```

Comment:
Probably the final thing here:  the `put(remote)` has do more than just make the state equal to the remote state. If this is a global db, it has to "merge" remote changes in without losing any siteLocations and if it is a storage site it has to update metadata without losing the storageLocation. 

See lines 615-634 in fenwick InventoryHarvester: since the scenario here is mismatched checksums you are doing more or less the `else` part of that.

Reply:
Updated ArtifactValidator to preserve the local Artifacts site and storage locations.